### PR TITLE
Sane default configuration for sabnzbd module

### PR DESCRIPTION
**Pull request by stephenmw**
Added option to set user. Use unprivileged user by default. Add sane default
for configuration location.

This is my first PR to nixpkgs. Go easy on me :-)

Also, this will probably break people currently using sabnzbd since it sets different permissions.

### CODE REVIEW

**Comment by edwtjo** [comment]
A minor nitpick is that you could also add a default group

File: nixos/modules/services/networking/sabnzbd.nix [hunk 1]
```diff
@@ -31,23 +37,21 @@ in
...
 
-    users.extraUsers =
-      [ { name = "sabnzbd";
+    users.extraUsers.sabnzbd = {
           uid = config.ids.uids.sabnzbd;
           description = "sabnzbd user";
-          home = "/homeless-shelter";
-        }
-      ];
+          home = "/var/lib/sabnzbd/";
```
Comment (edwtjo) [inline nixos/modules/services/networking/sabnzbd.nix]: also add `createHome = true;`

**Comment by stephenmw** [comment]
Should the default group be named sabnzbd?

**Comment by edwtjo** [comment]
I think there is one in ids with that name already, it's just commented out, use that one.

**Comment by stephenmw** [comment]
So, there wasn't an id already there. I am concerned because the uid and gid are different.

**Comment by edwtjo** [comment]
I must have been dreaming again.. Anyway there isn't a formed consensus on these things. I personally don't care that the integer value for the group doesn't match up with the uid value. I care that the default username and groupname match the servicename and that there is a default. The important thing is that the value isn't just changed on rut. If you decide to change a uid/gid value then you need to migrate previous installations using something like an activation chown/chmod checker. This doens't seem to be the issue here since you're adding a new group.

**Comment by edwtjo** [comment]
Looking good, thanks!

**Pull request merged**